scripts/controlnet.py

```diff
@@ -2,37 +2,37 @@
 import os
 from collections import OrderedDict
 from copy import copy
-from typing import Union, Dict, Optional, List
+from typing import Dict, Optional
 import importlib
 import modules.scripts as scripts
 from modules import shared, devices, script_callbacks, processing, masking, images
 import gradio as gr
 
 from einops import rearrange
-from scripts import global_state, hook, external_code, processor, batch_hijack, controlnet_version
+from scripts import global_state, hook, external_code, processor, batch_hijack, controlnet_version, utils
 importlib.reload(processor)
 importlib.reload(global_state)
 importlib.reload(hook)
 importlib.reload(external_code)
 importlib.reload(batch_hijack)
+importlib.reload(utils)
```

Comment:
Most likely we want to reload `utils` before `global_state`, as global_state.py imports `utils`. Sometimes python does not complain about these, but we may have a harder time decoupling them in the future.

Reply:
Actually, I'm a bit confused. There seems to be a circular import between global_state.py and utils.py, can we move some of the code to remove the cycle? Ideally we want the dependency graph between our files to have a directed acyclic graph structure.

Reply:
Done.